### PR TITLE
Improve metrics models

### PR DESCRIPTION
**Pull request by johnnv1**
Updates:
- alter from property to cached_property the metrics models
  - at tests/evaluation/model/test_BinaryClassificationMetrics.py
    - Move the FP, TP, FN, TN for the instantiation moment
  - lapixdl/evaluation/model.py
- Fix the division by zero at IoU
  - at lapixdl/evaluation/model.py
    - at class BinarySegmentationMetrics
    - return math.nan if quocient is zero
- add to_dict method at models
  - at lapixdl/evaluation/model.py
  - add to all class an method to get the metrics values in dictonary format
- remove unused imports
  - at tests/evaluation/evaluate/test_evaluate_detection.py
- add pandas as requeriments
  - at setup.py
-  add to_dataframe into multicategorical models
  - at lapixdl/evaluation/model.py
  - Create a function to convert an cls.by_class into a dict
  - add method  to convert the to_dict into an Pandas DataFrame
-  update print metrics with to_dataframe
  - at lapixdl/evaluation/evaluate.py
  - Use the to_dataframe into the print metrics

### CODE REVIEW

**Comment by johnnv1** [comment]
# bug
- `@chached_property` just work for python > 3.8, colab is at 3.7. The solution is to use `@property` and `@lru_cache` that was the same as `@chached_property` and work for python > 3.2

File: lapixdl/evaluation/model.py
Comment (Andrevmatias) [inline lapixdl/evaluation/model.py:83]: Cool! Didn't know this annotation 🚀

**Review by Andrevmatias — APPROVED** [review]

**Comment by johnnv1** [comment]
@Andrevmatias we can't merge this, when using lru_cache at class we have some hash problems, i forgot to fix that

**Pull request merged**
